app/frontend/src/features/auth/signup/BasicForm.tsx

```diff
@@ -0,0 +1,109 @@
+import { InputLabel } from "@material-ui/core";
+import Button from "components/Button";
+import TextField from "components/TextField";
+import { useAuthContext } from "features/auth/AuthProvider";
+import useAuthStyles from "features/auth/useAuthStyles";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { service } from "service";
+import {
+  emailValidationPattern,
+  lowercaseAndTrimField,
+  nameValidationPattern,
+} from "utils/validation";
+
+import {
+  CONTINUE,
+  EMAIL_EMPTY,
+  EMAIL_LABEL,
+  EMAIL_REQUIRED,
+  NAME_EMPTY,
+  NAME_LABEL,
+  NAME_REQUIRED,
+} from "../constants";
+
+type SignupBasicInputs = {
+  name: string;
+  email: string;
+};
+
+export default function BasicForm() {
+  const { authActions } = useAuthContext();
+  const authClasses = useAuthStyles();
+  const [loading, setLoading] = useState(false);
+
+  const { register, handleSubmit, errors } = useForm<SignupBasicInputs>({
+    mode: "onBlur",
+    shouldUnregister: false,
+  });
+
+  const onSubmit = handleSubmit(async (data: SignupBasicInputs) => {
+    setLoading(true);
+    authActions.clearError();
+    try {
+      const sanitizedEmail = lowercaseAndTrimField(data.email);
+      authActions.updateSignupState(
+        await service.auth.startSignup(data.name, sanitizedEmail)
+      );
+    } catch (err) {
+      authActions.authError(err.message);
+    }
+    setLoading(false);
+  });
+
+  return (
+    <>
+      <form className={authClasses.form} onSubmit={onSubmit}>
+        <InputLabel className={authClasses.formLabel} htmlFor="name">
```

Comment:
This doesn't work for accessibility. I think the TextField has to be an Input and they have to be wrapped with FormControl.

Reply:
I just check this one, but probably the same applies to all the signup and login ones. That's probably why the forum user said we aren't accessible! Would be good to fix this now.

Reply:
I was wrong about this, it's fine